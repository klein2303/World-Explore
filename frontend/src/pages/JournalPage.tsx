import { Link, useParams } from "react-router-dom";
import Navbar from "../components/Navbar/Navbar";
import { PiArrowElbowDownLeft } from "react-icons/pi";
import styles from "../styles/JournalPage.module.css"
import ReviewBox from "../components/ReviewBox/ReviewBox";
import { SlPencil } from "react-icons/sl";
import { useEffect } from "react";


const JournalPage = () => {
    // Get countryName from the URL 
    const { countryName } = useParams<{ countryName: string }>();
    
    // Let country be an default string if countryName is empty 
    const country = countryName || "";

    // Scroll to top when the component mounts
    useEffect(() => {
        window.scrollTo({ top: 0});
      }, []);

    return (
        <>
            <Navbar/>
            <main className= {styles.page} aria-label= "Page for reading your journals">
                <Link to = {"/MyJournals"} className= {styles.returnLink} aria-label= "Return to all journals"> <PiArrowElbowDownLeft /> Return to all journals</Link>
                <section className= {styles.upperSection}>
                    <p className={styles.title}>My {countryName} journals</p>
                    <button className= {styles.addButton} aria-label= "Add new journal entry">Add new journal entry <SlPencil /></button>
                </section>
                <ReviewBox country={country}  />
            </main>
        </>
    );
};

export default JournalPage;
