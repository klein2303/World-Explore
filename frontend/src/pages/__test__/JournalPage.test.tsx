// JournalPage.test.tsx

import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import JournalPage from '../JournalPage';
import { useParams } from 'react-router-dom';
import { describe, it, vi } from 'vitest';

// Mocking useParams to avoid actual route dependency
vi.mock('react-router-dom', async () => {
    const actual = await vi.importActual('react-router-dom');
    return {
      ...actual,
      useParams: vi.fn(),
    };
  });
  
  describe('JournalPage', () => {
    // Test with a countryName provided
    it('renders correctly with a countryName', () => {
      vi.mocked(useParams).mockReturnValue({ countryName: 'Japan' });
  
      const { asFragment } = render(
        <MemoryRouter>
          <JournalPage />
        </MemoryRouter>
      );
  
      // Check if the correct title is rendered using aria-label
      expect(screen.getByLabelText('Title')).toHaveTextContent(/My Japan journals/i);
      // Check if the add button is rendered using aria-label
      expect(screen.getByLabelText('Add new journal entry')).toBeInTheDocument();
      // Check if the return link is rendered using aria-label
      expect(screen.getByLabelText('Return to all journals')).toBeInTheDocument();
  
      // Snapshot test
      expect(asFragment()).toMatchSnapshot();
    });
  
    // Test without a countryName
    it('renders correctly without a countryName', () => {
      vi.mocked(useParams).mockReturnValue({ countryName: undefined });
  
      const { asFragment } = render(
        <MemoryRouter>
          <JournalPage />
        </MemoryRouter>
      );
  
      // Check if the default "My journals" title is rendered using aria-label
      expect(screen.getByLabelText('Title')).toHaveTextContent(/My journals/i);
      // Check if the add button is rendered using aria-label
      expect(screen.getByLabelText('Add new journal entry')).toBeInTheDocument();
      // Check if the return link is rendered using aria-label
      expect(screen.getByLabelText('Return to all journals')).toBeInTheDocument();
  
      // Snapshot test
      expect(asFragment()).toMatchSnapshot();
    });
  });