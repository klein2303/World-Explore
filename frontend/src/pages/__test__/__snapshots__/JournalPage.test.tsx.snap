// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JournalPage > renders correctly with a countryName 1`] = `
<DocumentFragment>
  <main>
    <nav
      aria-label="Main Navigation"
      class="_navbar_278c00"
      role="navigation"
    >
      <svg
        aria-label="Open Mobile Menu"
        class="_hamburgmenu_278c00"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <a
        class="_navTitle_278c00"
        href="/"
      >
        <p>
          WorldExplore
        </p>
      </a>
      <div
        aria-label="Navigation Links"
        class="_navlinks_278c00"
        role="group"
      >
        <a
          class="_navlink_278c00"
          href="/ExploreCountries"
        >
          <svg
            aria-hidden="true"
            class="_search_278c00"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.25 2a8.25 8.25 0 0 1 6.34 13.53l5.69 5.69a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-5.69-5.69A8.25 8.25 0 1 1 10.25 2ZM3.5 10.25a6.75 6.75 0 1 0 13.5 0 6.75 6.75 0 0 0-13.5 0Z"
            />
          </svg>
          <p>
            Explore Countries
          </p>
        </a>
        <a
          class="_navlink_278c00"
          href="/MyJournals"
        >
          <svg
            aria-hidden="true"
            class="_pin_278c00"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 10c0 6-8 12-8 12s-8-6-8-12a8 8 0 0 1 16 0Z"
            />
            <circle
              cx="12"
              cy="10"
              r="3"
            />
          </svg>
          <p>
            My Journals
          </p>
        </a>
        <svg
          aria-label="Open Profile Menu"
          class="_profile_278c00"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5C14.2091 5 16 6.79086 16 9ZM14 9C14 10.1046 13.1046 11 12 11C10.8954 11 10 10.1046 10 9C10 7.89543 10.8954 7 12 7C13.1046 7 14 7.89543 14 9Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM3 12C3 14.0902 3.71255 16.014 4.90798 17.5417C6.55245 15.3889 9.14627 14 12.0645 14C14.9448 14 17.5092 15.3531 19.1565 17.4583C20.313 15.9443 21 14.0524 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12ZM12 21C9.84977 21 7.87565 20.2459 6.32767 18.9878C7.59352 17.1812 9.69106 16 12.0645 16C14.4084 16 16.4833 17.1521 17.7538 18.9209C16.1939 20.2191 14.1881 21 12 21Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </nav>
  </main>
  <main
    aria-label="Page for reading your journals"
    class="_page_51c712"
  >
    <a
      aria-label="Return to all journals"
      class="_returnLink_51c712"
      href="/MyJournals"
    >
       
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M200,32V176a8,8,0,0,1-8,8H67.31l34.35,34.34a8,8,0,0,1-11.32,11.32l-48-48a8,8,0,0,1,0-11.32l48-48a8,8,0,0,1,11.32,11.32L67.31,168H184V32a8,8,0,0,1,16,0Z"
        />
      </svg>
       Return to all journals
    </a>
    <section
      class="_upperSection_51c712"
    >
      <p
        aria-label="Title"
        class="_title_51c712"
      >
        My Japan journals
      </p>
      <button
        aria-label="Add new journal entry"
        class="_addButton_51c712"
      >
        Add new journal entry 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M964.256 49.664C929.392 16.256 890.933-.672 849.877-.672c-64.192 0-111.024 41.472-123.841 54.176-18.032 17.856-633.152 633.2-633.152 633.2a33.011 33.011 0 0 0-8.447 14.592C70.565 752.559 1.077 980.016.387 982.304c-3.567 11.648-.384 24.337 8.208 32.928a32.336 32.336 0 0 0 22.831 9.44c3.312 0 6.655-.496 9.919-1.569 2.352-.767 237.136-76.655 275.775-88.19a32.736 32.736 0 0 0 13.536-8.033c24.416-24.128 598.128-591.456 636.208-630.784 39.392-40.592 58.96-82.864 58.208-125.616-.784-42.208-21.248-82.848-60.816-120.816zM715.845 155.84c16.304 3.952 54.753 16.862 94.017 56.479 39.68 40.032 50.416 85.792 52.416 96.208-125.824 125.168-415.456 411.728-529.632 524.672-10.544-24.56-27.584-54.144-54.993-81.76-33.471-33.728-67.536-52.783-93.808-63.503 112.992-113.008 408.08-408.224 532-532.096zM140.39 741.95c17.584 4.672 54.111 18.224 91.344 55.76 28.672 28.912 42.208 60.8 48.288 80.24-44.48 14.304-141.872 47.92-203.76 67.872 18.336-60.336 49.311-154.304 64.128-203.872zm780.031-491.584a1748.764 1748.764 0 0 1-6.065 6.16c-10.113-26.049-27.857-59.52-58.577-90.496-31.391-31.648-63.231-50.32-88.75-61.36 2.175-2.16 3.855-3.857 4.511-4.496 3.664-3.617 36.897-35.376 78.32-35.376 23.84 0 47.248 10.88 69.617 32.32 26.511 25.424 40.175 50.512 40.624 74.592.431 24.576-12.913 51.04-39.68 78.656z"
          />
        </svg>
      </button>
    </section>
    <main>
      <section
        aria-label="Section containing the journal entry"
        class="_reviewBox_756783"
      >
        <p
          class="_reviewTitle_756783"
        >
          Cherry Blossom Festival
        </p>
        <p
          class="_date_756783"
        >
          2024.04.05
        </p>
        <article
          class="_reviewText_756783"
        >
          Attending the cherry blossom festival was one of the most serene experiences ever. 
        </article>
        <section
          aria-label="Shows rating of a trip, illustrated as stars"
          class="_starsContainer_756783"
        >
          <svg
            aria-label="filled star"
            class="_starFilled_756783"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <svg
            aria-label="filled star"
            class="_starFilled_756783"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <svg
            aria-label="filled star"
            class="_starFilled_756783"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <svg
            aria-label="filled star"
            class="_starFilled_756783"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <svg
            aria-label="filled star"
            class="_starFilled_756783"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
        </section>
        <section
          aria-label="Change the privacy of your journal"
          class="_publicContainer_756783"
        >
          <p>
            Make this journal entry public
          </p>
          <svg
            aria-label="Make journey private"
            class="_checkBox_756783"
            fill="currentColor"
            height="1em"
            role="button"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V5h14v14zM17.99 9l-1.41-1.42-6.59 6.59-2.58-2.57-1.42 1.41 4 3.99z"
            />
          </svg>
        </section>
      </section>
      <section
        aria-label="Section containing the journal entry"
        class="_reviewBox_756783"
      >
        <p
          class="_reviewTitle_756783"
        >
          Amazing Sushi
        </p>
        <p
          class="_date_756783"
        >
          2023.11.10
        </p>
        <article
          class="_reviewText_756783"
        >
          The sushi in Japan is unparalleled! Highly recommend visiting Tsukiji Market. 
        </article>
        <section
          aria-label="Shows rating of a trip, illustrated as stars"
          class="_starsContainer_756783"
        >
          <svg
            aria-label="filled star"
            class="_starFilled_756783"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <svg
            aria-label="filled star"
            class="_starFilled_756783"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <svg
            aria-label="filled star"
            class="_starFilled_756783"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <svg
            aria-label="filled star"
            class="_starFilled_756783"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <svg
            aria-label="empty star"
            class="_starNotFilled_756783"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
            />
          </svg>
        </section>
        <section
          aria-label="Change the privacy of your journal"
          class="_publicContainer_756783"
        >
          <p>
            Make this journal entry public
          </p>
          <svg
            aria-label="Make journey public"
            class="_checkBox_756783"
            fill="currentColor"
            height="1em"
            role="button"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </section>
      </section>
    </main>
  </main>
</DocumentFragment>
`;

exports[`JournalPage > renders correctly without a countryName 1`] = `
<DocumentFragment>
  <main>
    <nav
      aria-label="Main Navigation"
      class="_navbar_278c00"
      role="navigation"
    >
      <svg
        aria-label="Open Mobile Menu"
        class="_hamburgmenu_278c00"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <a
        class="_navTitle_278c00"
        href="/"
      >
        <p>
          WorldExplore
        </p>
      </a>
      <div
        aria-label="Navigation Links"
        class="_navlinks_278c00"
        role="group"
      >
        <a
          class="_navlink_278c00"
          href="/ExploreCountries"
        >
          <svg
            aria-hidden="true"
            class="_search_278c00"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.25 2a8.25 8.25 0 0 1 6.34 13.53l5.69 5.69a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-5.69-5.69A8.25 8.25 0 1 1 10.25 2ZM3.5 10.25a6.75 6.75 0 1 0 13.5 0 6.75 6.75 0 0 0-13.5 0Z"
            />
          </svg>
          <p>
            Explore Countries
          </p>
        </a>
        <a
          class="_navlink_278c00"
          href="/MyJournals"
        >
          <svg
            aria-hidden="true"
            class="_pin_278c00"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 10c0 6-8 12-8 12s-8-6-8-12a8 8 0 0 1 16 0Z"
            />
            <circle
              cx="12"
              cy="10"
              r="3"
            />
          </svg>
          <p>
            My Journals
          </p>
        </a>
        <svg
          aria-label="Open Profile Menu"
          class="_profile_278c00"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5C14.2091 5 16 6.79086 16 9ZM14 9C14 10.1046 13.1046 11 12 11C10.8954 11 10 10.1046 10 9C10 7.89543 10.8954 7 12 7C13.1046 7 14 7.89543 14 9Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM3 12C3 14.0902 3.71255 16.014 4.90798 17.5417C6.55245 15.3889 9.14627 14 12.0645 14C14.9448 14 17.5092 15.3531 19.1565 17.4583C20.313 15.9443 21 14.0524 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12ZM12 21C9.84977 21 7.87565 20.2459 6.32767 18.9878C7.59352 17.1812 9.69106 16 12.0645 16C14.4084 16 16.4833 17.1521 17.7538 18.9209C16.1939 20.2191 14.1881 21 12 21Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </nav>
  </main>
  <main
    aria-label="Page for reading your journals"
    class="_page_51c712"
  >
    <a
      aria-label="Return to all journals"
      class="_returnLink_51c712"
      href="/MyJournals"
    >
       
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M200,32V176a8,8,0,0,1-8,8H67.31l34.35,34.34a8,8,0,0,1-11.32,11.32l-48-48a8,8,0,0,1,0-11.32l48-48a8,8,0,0,1,11.32,11.32L67.31,168H184V32a8,8,0,0,1,16,0Z"
        />
      </svg>
       Return to all journals
    </a>
    <section
      class="_upperSection_51c712"
    >
      <p
        aria-label="Title"
        class="_title_51c712"
      >
        My  journals
      </p>
      <button
        aria-label="Add new journal entry"
        class="_addButton_51c712"
      >
        Add new journal entry 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M964.256 49.664C929.392 16.256 890.933-.672 849.877-.672c-64.192 0-111.024 41.472-123.841 54.176-18.032 17.856-633.152 633.2-633.152 633.2a33.011 33.011 0 0 0-8.447 14.592C70.565 752.559 1.077 980.016.387 982.304c-3.567 11.648-.384 24.337 8.208 32.928a32.336 32.336 0 0 0 22.831 9.44c3.312 0 6.655-.496 9.919-1.569 2.352-.767 237.136-76.655 275.775-88.19a32.736 32.736 0 0 0 13.536-8.033c24.416-24.128 598.128-591.456 636.208-630.784 39.392-40.592 58.96-82.864 58.208-125.616-.784-42.208-21.248-82.848-60.816-120.816zM715.845 155.84c16.304 3.952 54.753 16.862 94.017 56.479 39.68 40.032 50.416 85.792 52.416 96.208-125.824 125.168-415.456 411.728-529.632 524.672-10.544-24.56-27.584-54.144-54.993-81.76-33.471-33.728-67.536-52.783-93.808-63.503 112.992-113.008 408.08-408.224 532-532.096zM140.39 741.95c17.584 4.672 54.111 18.224 91.344 55.76 28.672 28.912 42.208 60.8 48.288 80.24-44.48 14.304-141.872 47.92-203.76 67.872 18.336-60.336 49.311-154.304 64.128-203.872zm780.031-491.584a1748.764 1748.764 0 0 1-6.065 6.16c-10.113-26.049-27.857-59.52-58.577-90.496-31.391-31.648-63.231-50.32-88.75-61.36 2.175-2.16 3.855-3.857 4.511-4.496 3.664-3.617 36.897-35.376 78.32-35.376 23.84 0 47.248 10.88 69.617 32.32 26.511 25.424 40.175 50.512 40.624 74.592.431 24.576-12.913 51.04-39.68 78.656z"
          />
        </svg>
      </button>
    </section>
    <main>
      <p
        class="_noJournal_756783"
      >
        No reviews found for 
      </p>
    </main>
  </main>
</DocumentFragment>
`;
